src/Aquifer.API/Endpoints/Resources/Content/SendForPublisherReview/Endpoint.cs

```diff
@@ -19,9 +20,44 @@ public override void Configure()
     public override async Task HandleAsync(Request request, CancellationToken ct)
     {
         var contentIds = request.ContentId is not null ? [request.ContentId.Value] : request.ContentIds!;
+        var draftVersions = await GetDraftVersions(contentIds, ct);
+
+        var user = await userService.GetUserFromJwtAsync(ct);
+        var isPublisher = user.Role == UserRole.Publisher;
+        foreach (var draftVersion in draftVersions)
+        {
+            await ValidateAssignedUser(user, request.AssignedUserId, draftVersion, ct);
+
+            var newStatus = GetNewStatus(isPublisher, draftVersion.ResourceContent.Status);
+
+            await historyService.AddSnapshotHistoryAsync(draftVersion,
+                draftVersion.AssignedUserId,
+                draftVersion.ResourceContent.Status,
+                ct);
+
+            draftVersion.ResourceContent.Status = newStatus;
+            await AssignUserToContent(newStatus, request.AssignedUserId, draftVersion, user, ct);
+
+            if (newStatus != draftVersion.ResourceContent.Status)
+            {
+                await historyService.AddStatusHistoryAsync(draftVersion, newStatus, user.Id, ct);
+            }
+
+            Helpers.SanitizeTiptapContent(draftVersion);
+        }
+
+        await dbContext.SaveChangesAsync(ct);
+
+        Response.ChangedByPublisher = isPublisher;
+    }
+
+    private async Task<List<ResourceContentVersionEntity>> GetDraftVersions(List<int> contentIds, CancellationToken ct)
```

Comment:
I think this is what the helper should be doing, where these `allowedStatuses` are passed in.